src/Chart.tsx

```diff
@@ -138,36 +134,26 @@ export const Chart = forwardRef<ChartHandle, ChartProps>(
 		const { controlledHoverSignal, selectedIdSignalName, selectedSeriesSignalName } = useSpecProps(spec);
 		const chartConfig = useMemo(() => getChartConfig(config, colorScheme), [config, colorScheme]);
 
-		// Need to de a deep copy of the data because vega tries to transform the data
-		const chartData = useMemo(() => {
-			const clonedData = JSON.parse(JSON.stringify(data));
-
-			// We received a full Vega data array with potentially multiple dataset objects
-			if (isVegaData(clonedData)) {
-				return extractValues(clonedData);
-			}
-
-			// We received a simple array of data and we'll set a default key of 'table' to reference internally
-			return { [TABLE]: clonedData };
-		}, [data]);
-
 		useEffect(() => {
 			const tooltipElement = document.getElementById('vg-tooltip-element');
 			if (!tooltipElement) return;
 			// Hide tooltips on all charts when a popover is open
 			tooltipElement.hidden = popoverIsOpen;
 
 			// if the popover is closed, reset the selected data
-			if (!popoverIsOpen) {
+			if (!popoverIsOpen && chartView.current) {
 				selectedData.current = null;
+
+				setTimeout(() => {
+					chartView.current?.runAsync();
+				}, 1000);
```

Comment:
Is this `1000` just an arbitrary number? Is it worth defining a constant somewhere if we expect to do this (timeouts) in more than one place?

Reply:
I did not mean to leave this in there haha